crates/ruff/src/autofix/mod.rs

```diff
@@ -50,7 +50,7 @@ fn apply_fixes<'a>(
     let mut fixed = FxHashMap::default();
     let mut source_map = SourceMap::default();
 
-    for (rule, fix) in diagnostics
+    'outer: for (rule, fix) in diagnostics
```

Comment:
Nit: If you want to avoid the labels, but it requires an additional vector: 

```rust
        let mut edits = fix
            .edits()
            .iter()
            .filter(|edit| !applied.contains(edit))
            .peekable();

        // Lazily enforce any isolation and positional requirements (e.g., avoid applying
        // overlapping fixes, but avoid applying this requirement if all fixes in the edit were
        // already applied).
        if let Some(first) = edits.peek() {
            // If this fix requires isolation, and we've already applied another fix in the
            // same isolation group, skip it.
            if let IsolationLevel::Group(id) = fix.isolation() {
                if !isolated.insert(id) {
                    continue;
                }
            }

            // Best-effort approach: if this fix overlaps with a fix we've already applied,
            // skip it.
            if last_pos.map_or(false, |last_pos| last_pos >= first.start()) {
                continue;
            }
        }

        for edit in edits {
            // Add all contents from `last_pos` to `fix.location`.
            let slice = locator.slice(TextRange::new(last_pos.unwrap_or_default(), edit.start()));
            output.push_str(slice);

            // Add the start source marker for the patch.
            source_map.push_start_marker(edit, output.text_len());

            // Add the patch itself.
            output.push_str(edit.content().unwrap_or_default());

            // Add the end source marker for the added patch.
            source_map.push_end_marker(edit, output.text_len());

            // Track that the edit was applied.
            last_pos = Some(edit.end());
            applied_edits.push(edit);
        }

        applied.extend(applied_edits.drain(..));
        *fixed.entry(rule).or_default() += 1;
    }
```

I don't have a preference to be honest. I just thought this is complicated, and then noticed that it is because we read from and write to `applied` at the same time.

Reply:
I think what you have here is a big clarity improvement, worth the allocation.